### PR TITLE
Terraforming!

### PR DESCRIPTION
**Pull request by yairm210**
The long *long* awaited feature
Tested by adding 
 "Create a [Forest] on this tile <for [0] movement>", "Turn this tile into a [Ocean] <for [0] movement>"

### CODE REVIEW

File: core/src/com/unciv/models/ruleset/unique/UniqueType.kt [hunk 1]
```diff
@@ -755,6 +755,9 @@ enum class UniqueType(
     SkipPromotion("Doing so will consume this opportunity to choose a Promotion", UniqueTarget.Promotion),
     FreePromotion("This Promotion is free", UniqueTarget.Promotion),
 
+    OneTimeCreateTerrainFeatureImprovement("Create a [terrainFeature/improvementName] on this tile", UniqueTarget.UnitTriggerable),
```
Comment (SeventhM) [inline core/src/com/unciv/models/ruleset/unique/UniqueType.kt:758]: Makes `ConstructImprovementInstantly` redundant? Tbh, I think some of the unit actions should probably become triggers anyways. Probably should deprecate?

Comment (yairm210) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:758]: To do that, we'll need to add a way to limit this construction according to the improvement's 'can be constructed on' rules

Comment (SeventhM) [reply core/src/com/unciv/models/ruleset/unique/UniqueType.kt:758]: To be fair, I'm not sure this unique implies it should ignore limitations

**Comment by SeventhM** [comment]
Why are we adding a new parameter in the first place and not using TerrainName? Especially if we want to support adding in nat wonders. I kinda feel like it should either be one or the other: Either allow Nat Wonders or just go the full distance and allowing adding terrain wholesale. You can already change the base terrain with some of the natural wonders anyways

**Comment by yairm210** [comment]
Good point regarding cities!

**Pull request merged**